lib/mwoffliner.lib.js

```diff
@@ -911,6 +927,99 @@ module.exports = {
                 }
             }
 
+            function storeDependencies( parsoidDoc, articleId, finished ) {
```

Comment:
kelson42, this function download and write module on disk only if required (ie the first time a module appears). If not required, there are no file system access at all : the check is done through redis.

The difference with what you say is that the modules are downloaded during the process instead of at the end, but there are no extra-download or extra file system access